src/macros.rs

```diff
@@ -0,0 +1,23 @@
+#[macro_export]
+macro_rules! increment_counter {
+    ($state:ident.$property:ident, $metric:ident) => {{
+        use {opentelemetry::Context, tracing::debug};
+
+        if let Some(metrics) = &$state.$property {
```

Comment:
Looks like this macro only supports a first argument in the form `ident.ident`? What if I wanted to pass a more deeply-nested identifier such as `ident.ident.ident`? Would using `$metrics:expr` as the first macro parameter make this more generalized?

Reply:
It can be added but right now I know that it will only be one deep -> can be looked at later though